src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,49 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(biteAnimal) {
+    const animalIsHerbivore = biteAnimal instanceof Herbivore;
+    const noHidden = biteAnimal.hidden === false;
+
+    if (animalIsHerbivore && noHidden) {
+      biteAnimal.health -= 50;
+    }
+
+    if (biteAnimal.health <= 0) {
+      Animal.alive.forEach((value, index) => {
+        if (value.health <= 0) {
+          Animal.alive.splice(index, 1);
```

Comment:
Using `splice` within a `forEach` loop to remove elements from an array can lead to unexpected behavior, especially if multiple animals have the same reference. Consider using the `filter` method to create a new array excluding the animals with `health <= 0`. This will ensure that the correct instances are removed from the `Animal.alive` array.